recipes/openvdb/8.1.0/conanfile.py

```diff
@@ -0,0 +1,202 @@
+        "simd": None,
+    }
+
+    _cmake = None
+
+    _compilers_min_version = {
+        "msvc": "19.10",
+        "Visual Studio": "15",  # Should we check toolset?
+        "gcc": "6.3.1",
+        "clang": "3.8",
+        "apple-clang": "3.8",
+        "intel": "17",
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _check_compilier_version(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if version < self._compilers_min_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a %s version greater than %s" % (self.name, compiler, self._compilers_min_version[compiler]))
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime:
+                raise ConanInvalidConfiguration("Visual Studio build for shared library with MT runtime is not supported")
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        if self.settings.arch not in ("x86", "x86_64"):
+            if self.options.simd:
+                raise ConanInvalidConfiguration("Only intel architectures support SSE4 or AVX.")
+        self._check_compilier_version()
+
+    def requirements(self):
+        self.requires("boost/1.76.0")
+        self.requires("tbb/2020.3")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_blosc:
+            self.requires("c-blosc/1.20.1")
+        if self.options.with_log4cplus:
+            self.requires("log4cplus/2.0.5")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _patch_sources(self):
+        # Remove FindXXX files from OpenVDB. Let Conan do the job
+        tools.remove_files_by_mask(os.path.join(self._source_subfolder, "cmake"), "Find*")
+        with open("FindBlosc.cmake", "w") as f:
+            f.write(
+                """find_package(c-blosc)
+if(c-blosc_FOUND)
+    add_library(blosc INTERFACE)
+    target_link_libraries(blosc INTERFACE c-blosc::c-blosc)
+    add_library(Blosc::blosc ALIAS blosc)
+endif()
+"""
+            )
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        # Force MD(d)
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["CMAKE_MSVC_RUNTIME_LIBRARY"] = "MultiThreaded$<$<CONFIG:Debug>:Debug>"
```

Comment:
Usually we don't need it. Is there are different configuration in openvdb which requires it?

Reply:
I was getting an error when building it as a static lib because they always default to MD(d) and when the test package tried to link there was a runtime mismatch
`error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in test_package.obj`
But there is a mistake. I forgot to add the condition to do so only when building the static lib.

Reply:
can this be removed since MT shared is blocked? But this check should be moved to validate (ie block all MT)